clients/cli/src/command.rs

```
@@ -2099,6 +2100,130 @@ async fn command_unwrap(
+
+        let account_state = StateWithExtensionsOwned::<Account>::unpack(account_data.data)?;
+
+        if let Some(amount) = amount {
+            if account_state.base.amount < amount {
+                return Err(format!(
+                    "Error: Sender has insufficient funds, current balance is {} SOL",
+                    build_balance_message(account_state.base.amount, false, false)
+                )
+                .into());
+            }
+        }
+
+        balance = Some(account_state.base.amount);
+
+        if !use_associated_account && account_state.base.mint != *token.get_address() {
+            return Err(format!("{} is not a native token account", source_account).into());
+        }
+
+        if config.rpc_client.get_balance(&destination_account).await? == 0 {
+            // if it doesn't exist, we gate transfer with a different flag
+            if allow_unfunded_recipient {
+                println_display(
+                    config,
+                    format!("Funding recipient: {}", destination_account,),
+                );
+
+                let rent_exempt_lamports = config
+                    .rpc_client
+                    .get_minimum_balance_for_rent_exemption(0)
+                    .await?;
+                let fee_payer = config.fee_payer()?;
+                let instruction = transfer(
+                    &fee_payer.pubkey(),
+                    &destination_account,
+                    rent_exempt_lamports,
+                );
+                let recent_blockhash = config.rpc_client.get_latest_blockhash().await?;
+                let transaction = Transaction::new_signed_with_payer(
+                    &[instruction],
+                    Some(&fee_payer.pubkey()),
+                    &[fee_payer],
+                    recent_blockhash,
+                );
+                config
+                    .rpc_client
+                    .send_and_confirm_transaction(&transaction)
+                    .await?;
```

Let's take this out. Doing a separate transfer transaction to fund rent-exemption is very sketchy. Just let the command fail if the lamport amount is too low.

---

Thanks, resolved [here](https://github.com/solana-program/token-2022/pull/857/commits/c90b2ff930f6cf9c7264f8c9a8ba6b0abf2d6d88)